clients/cloud/nodejs/producer.js

```diff
@@ -0,0 +1,57 @@
+const Kafka = require('node-rdkafka');
+const { configFromCli } = require('./config');
+
+async function createProducer(config) {
+  const producer = new Kafka.Producer({
+    'metadata.broker.list': config['bootstrap.servers'],
+    'sasl.username': config['sasl.username'],
+    'sasl.password': config['sasl.password'],
+    'security.protocol': 'SASL_SSL',
+    'sasl.mechanisms': 'PLAIN',
+    'dr_cb': true
+  });
+
+  return new Promise((resolve, reject) => {
+    producer
+      .on('ready', () => resolve(producer))
+      .on('event.error', (err) => {
+        console.warn('event.error', err);
+        reject(err);
+    });
+
+    producer.connect();
+  });
+}
+
+function range(length) {
+  return Array.from(Array(length).keys());
```

Comment:
I don't really know nodejs, but is creating two arrays really the most effective way to iterate over 10 numbers?

Reply:
Yeah, there are different ways to do this for sure. The outer `Array.from` can be avoided if we reuse the iterator returned by `keys()` and consume using a `for...of` loop. Debatably, the usage of the latter is more exotic in JavaScript-land, compared to the `forEach` alternative. Of course there's a good argument for a standard `while` loop, too. 

I'll push the `for...of` option next, so we can feel it out.